Editor/Extensions/SerializedObjectExtensions.cs

```diff
@@ -1,10 +1,53 @@
+using System;
 using System.Collections.Generic;
+using RoyTheunissen.AssetPalette.Windows;
 using UnityEditor;
 
 namespace RoyTheunissen.AssetPalette.Extensions
 {
     public static partial class SerializedObjectExtensions
     {
+        public static SerializedProperty FindPropertyFromId(
+            this SerializedObject serializedObject, string guid, string rootCollectionPath)
```

Comment:
I just realized, this implementation iterates over every child in the collection until it finds the one with the right guid, which is not very efficient.

If you structure the path like "`GUID`/`GUID`/`GUID`" then you can first find the first GUID on the root level, then you find the next GUID in its direct children, then you find the next GUID in its direct children, and you don't have to go down into hierarchies that aren't in the GUID path. That was my intention for that formatting.

Then you do have to do a bit more work to get the path from a serialized property, because you have to recurse upwards, but that is not so costly because you don't travel up any elements you don't need to. You can see an example of this in: `SerializedPropertyExtensions.GetReferenceIdPathRecursive`

Reply:
I'm making some re-usable extension method for something like this, I think you'll be able to use that one